core/src/main/scala/zio/temporal/workflow/ZWorkflowQueue.scala

```diff
@@ -0,0 +1,63 @@
+package zio.temporal.workflow
+
+import io.temporal.workflow.{QueueConsumer, WorkflowQueue}
+import zio.Duration
+
+final class ZWorkflowQueue[E] private[zio] (val toJava: WorkflowQueue[E])
+    extends ZQueueConsumer[E](toJava)
+    with ZQueueProducer[E] {
+
+  override def offer(e: E): Boolean =
+    toJava.offer(e)
+
+  override def put(e: E): Unit =
+    toJava.put(e)
+
+  override def cancellablePut(e: E): Unit =
+    toJava.cancellablePut(e)
+
+  override def offer(e: E, timeout: zio.Duration): Boolean =
+    toJava.offer(e, timeout)
+
+  override def cancellableOffer(e: E, timeout: zio.Duration): Boolean =
+    toJava.cancellableOffer(e, timeout)
+}
+
+trait ZQueueProducer[E] {
+
+  def offer(e: E): Boolean
+
+  def put(e: E): Unit
+
+  def cancellablePut(e: E): Unit
+
+  def offer(e: E, timeout: Duration): Boolean
+
+  def cancellableOffer(e: E, timeout: Duration): Boolean
+
+}
+
+class ZQueueConsumer[E] private[zio] (toJava: QueueConsumer[E]) {
```

Comment:
Minor: can it be an interface as well? The ZWorkflowQueue can implement two interfaces

Reply:
Unfortunately, it should be a class because the original `QueueConsumer.map` returns QueueConsumer, which is a supertype for `WorkflowQueue`. That means we can't implement `map` method like:
```
def map[R](f: E => R): ZWorkflowQueue[R] =
    new ZWorkflowQueue(toJava.map(e => f(e)) // compiler error, toJava.map returns QueueConsumer but ZWorkflowQueue receives WorkflowQueue
```

Hope it makes sense.

Reply:
Okay, that's strange, but we should follow it 😄